src/main/fc/fc_core.c

```diff
@@ -434,6 +495,66 @@ void processRx(timeUs_t currentTimeUs)
         pidStabilisationState(PID_STABILISATION_ON);
     }
 
+    // If runaway_takeoff_prevention is enabled, accumulate the amount of time that throttle
+    // is above runaway_takeoff_deactivate_throttle with the any of the R/P/Y sticks deflected
+    // to at least runaway_takeoff_stick_percent percent while the pidSum on all axis is kept low.
+    // Once the amount of accumulated time exceeds runaway_takeoff_deactivate_delay then disable
+    // prevention for the remainder of the battery.
+
+    if (ARMING_FLAG(ARMED)
+        && pidConfig()->runaway_takeoff_prevention
+        && !runawayTakeoffCheckDisabled
+        && !STATE(FIXED_WING)) {
+
+        // Determine if we're in "flight"
+        //   - motors running
+        //   - throttle over runaway_takeoff_deactivate_throttle_percent
+        //   - sticks are active and have deflection greater than runaway_takeoff_deactivate_stick_percent
+        //   - pidSum on all axis is less then runaway_takeoff_deactivate_pidlimit
+        inFlight = false;
+        if (!feature(FEATURE_MOTOR_STOP) || isAirmodeActive() || (calculateThrottleStatus() != THROTTLE_LOW)) { // are motors running?
+            float pidSumLimit = pidConfig()->runaway_takeoff_deactivate_pidlimit * 10.0f;
+            if ((calculateThrottlePercent() >= pidConfig()->runaway_takeoff_deactivate_throttle)
+                && areSticksActive(pidConfig()->runaway_takeoff_deactivate_stick)
+                && (fabsf(axisPID_P[FD_PITCH] + axisPID_I[FD_PITCH] + axisPID_D[FD_PITCH]) < pidSumLimit)
+                && (fabsf(axisPID_P[FD_ROLL] + axisPID_I[FD_ROLL] + axisPID_D[FD_ROLL]) < pidSumLimit)
+                && (fabsf(axisPID_P[FD_YAW] + axisPID_I[FD_YAW]) < pidSumLimit)) {
+                
+                inFlight = true;
+                if (runawayTakeoffDeactivateUs == 0) {
+                    runawayTakeoffDeactivateUs = currentTimeUs;
```

Comment:
set `runawayTakeoffDeactivateUs = currentTimeUs + pidConfig()->runaway_takeoff_deactivate_delay * 1000;`
Then later your check can be simplified.

Reply:
See later comment, but I don't think this would work because of the needed accumulation logic.  However I think I can adopt this kind of logic for the detection phase inside of `static void subTaskPidController(timeUs_t currentTimeUs)`.  It's using a simple re-settable time limit.